src/main/webapp/app/course/dashboards/assessment-dashboard/assessment-dashboard-information.component.ts

```diff
@@ -1,30 +1,89 @@
+import { LegendPosition } from '@swimlane/ngx-charts';
+import { TranslateService } from '@ngx-translate/core';
+
+export class AssessmentDashboardInformationEntry {
+    constructor(public total: number, public tutor: number, public done?: number) {}
+
+    get doneToTotalPercentage(): string {
+        if (this.done == undefined) {
+            return '';
+        }
+
+        if (this.total === 0) {
+            return '100%';
+        }
+
+        return `${((100 * this.done) / this.total).toFixed().toString()}%`;
+    }
+}
 
 @Component({
     selector: 'jhi-assessment-dashboard-information',
     templateUrl: './assessment-dashboard-information.component.html',
 })
-export class AssessmentDashboardInformationComponent {
+export class AssessmentDashboardInformationComponent implements OnInit {
     @Input() isExamMode: boolean;
-    @Input() numberOfTutorAssessments: number;
     @Input() courseId: number;
+    @Input() examId?: number;
+    @Input() tutorId: number;
+    @Input() isAtLeastInstructor: boolean;
+
     @Input() complaintsEnabled: boolean;
     @Input() feedbackRequestEnabled: boolean;
-    @Input() tutorId: number;
-    @Input() numberOfTutorComplaints: number;
-    @Input() numberOfTutorMoreFeedbackRequests: number;
-    @Input() numberOfAssessmentLocks: number;
-    @Input() totalNumberOfAssessmentLocks: number;
 
-    @Input() examId?: number;
+    @Input() numberOfCorrectionRounds: number;
+    @Input() numberOfAssessmentsOfCorrectionRounds: DueDateStat[];
 
     @Input() totalNumberOfAssessments: DueDateStat;
     @Input() numberOfSubmissions: DueDateStat;
-    @Input() numberOfCorrectionRounds: number;
+    @Input() numberOfTutorAssessments: number;
     @Input() totalAssessmentPercentage: number;
-    @Input() numberOfAssessmentsOfCorrectionRounds: DueDateStat[];
-    @Input() isAtLeastInstructor: boolean;
-    @Input() numberOfComplaints: number;
-    @Input() numberOfMoreFeedbackRequests: number;
+
+    @Input() complaints: AssessmentDashboardInformationEntry;
+    @Input() moreFeedbackRequests: AssessmentDashboardInformationEntry;
+    @Input() assessmentLocks: AssessmentDashboardInformationEntry;
+    @Input() ratings: AssessmentDashboardInformationEntry;
+
+    // Graph data.
+    completedAssessmentsTitle: string;
+    openedAssessmentsTitle: string;
+    get assessments(): any[] {
+        return [
+            {
+                name: this.completedAssessmentsTitle,
+                value: this.totalNumberOfAssessments.total,
+            },
+            {
+                name: this.openedAssessmentsTitle,
+                value: this.numberOfSubmissions.total - this.totalNumberOfAssessments.total,
+            },
+        ];
+    }
+    view: [number, number] = [320, 150];
+    legendPosition = LegendPosition.Below;
+    get customColors(): any[] {
+        return [
+            {
+                name: this.completedAssessmentsTitle,
+                value: '#98C7EF',
```

Comment:
this fixed colors might be moved/defined another place, its just a minor thing

Reply:
This color is not shared with other components so I decided to put them inline. I can extract them into a constant, but I don't think `color1` and `color2` constants would add much value.